src/plugins/ui_stateful/edition.ts

```diff
@@ -60,19 +95,30 @@ export class EditionPlugin extends UIPlugin {
     "getCurrentEditedCell",
   ] as const;
 
-  private col: HeaderIndex = 0;
-  private row: HeaderIndex = 0;
-  private mode: EditionMode = "inactive";
-  private sheetId: UID = "";
-  private currentContent: string = "";
-  private currentTokens: EnrichedToken[] = [];
-  private selectionStart: number = 0;
-  private selectionEnd: number = 0;
-  private selectionInitialStart: number = 0;
-  private initialContent: string | undefined = "";
-  private previousRef: string = "";
-  private previousRange: Range | undefined = undefined;
-  private colorIndexByRange: { [xc: string]: number } = {};
+  private state!: State;
+
+  constructor(config: UIPluginConfig) {
+    super(config);
+    this.setDefaultState();
+  }
+
+  setDefaultState() {
```

Comment:
Isn't it possible to use the "values" from the EmptyState interface here (something like `this.state = new EmptyState` ? This would avoid to write all the values when reinitialising this.state. But this is more like a personnal relfexion, as I'm totally not sure if this can be done